src/common/ast-path.js

```diff
@@ -15,6 +15,24 @@ function getNodeStackIndexHelper(stack, count) {
   return -1;
 }
 
+const isThenable = (object) => typeof object?.then === "function";
```

Comment:
We should use `isPromise` ?

```js
const isPromise = (value) => value instanceof Promise;
```

`Promise` also has methods other than `then`.

Reply:
Was renamed during refactory, I think `isThenable` is more correct.

https://github.com/prettier/prettier/pull/13104/commits/f47256d4f62c670bf562cfdb66fe98f992398dcd

Reply:
`Promise` has methods other than `then`, such as `catch` and `finally`. Actually we use `Promise.prototype.finally` in `tryFinally` function.

Reply:
It don't has to be `Promise`, yes we use finally so it requires a Thenable with `finally` which I think it's fine, but people may use bluebird, which is faster than native Promise.

Reply:
I understood it doesn't has to be Promise. But, in our implementation, if `fn` (arguments of tryFinally) is a `{then: () => {}, finally: null }`, runtime exception will be occureed?

Reply:
Yes, and that's expected, 

> so it requires a Thenable with finally